### PR TITLE
Advisor and recommended actions swagger

### PR DESCRIPTION
**Pull request by tomapaunovic**
Adding swagger for advisor API on database/server and recommendations API (under individual advisors) on database level

### CODE REVIEW

**Comment by sarangan12** [comment]
@tomapaunovic 

Please take a look at the following jobs.

https://travis-ci.org/Azure/azure-rest-api-specs/jobs/245343147
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/245343148

The errors must be resolved before moving forward.

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (sarangan12) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: The error is here... This file is not a valid json file. Fix the error. Only then our linter will process the file and report the issues correctly

**Review by sarangan12 — CHANGES_REQUESTED** [review]

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (sarangan12) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: There is an extra comma here which makes the json ivalid

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (sarangan12) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: There is a missing comma here which makes the json invalid

**Review by sarangan12 — CHANGES_REQUESTED** [review]
Please fix the errors to ensure the JSON is valid. Once you generate/create the json you could try it in https://jsonlint.com/ to make sure the json is valid. Only then our validators will be able to run the file

**Comment by sarangan12** [comment]
@tomapaunovic 

Regarding your PR Please take a look at the build: https://travis-ci.org/Azure/azure-rest-api-specs/builds/246104656?utm_source=github_status&utm_medium=notification  
Especially, https://travis-ci.org/Azure/azure-rest-api-specs/jobs/246104665
There are several issues with your examples which must be resolved.

**Comment by sarangan12** [comment]
If you haveany issues resolving them, please come to our office hours

**Comment by jaredmoo** [comment]
@nathannfan

**Comment by jaredmoo** [comment]
Node, ruby, and python builds are broken, seems like an infra issue.

@tomapaunovic the following warnings in https://travis-ci.org/Azure/azure-rest-api-specs/jobs/247222965 can be ignored:
- OperationsAPIImplementation
- TrackedResourceListByImmediateParent
- EnumInsteadOfBoolean

The following should be fixed:
- XmsEnumValidation
- PatchInOperationName

**Comment by jaredmoo** [comment]
Just a couple more warnings to fix:
- https://travis-ci.org/Azure/azure-rest-api-specs/jobs/247428039#L1391
- https://travis-ci.org/Azure/azure-rest-api-specs/jobs/247428041#L1310

File: arm-sql/2015-05-01-preview/examples/DatabaseAdvisorGet.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/examples/DatabaseAdvisorGet.json]: These examples will show up on https://docs.microsoft.com , I suggest replacing real subscription id's with fake ones like in other examples

File: arm-sql/2015-05-01-preview/examples/DatabaseRecommendedActionGet.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/examples/DatabaseRecommendedActionGet.json]: can this name be simplified for documentation purposes?

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Use lower case consistently, e.g. `database advisors`

Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: It matters because this help text will be visible in https://docs.microsoft.com and Azure CLI command descriptions

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Server*s*_GetAdvisor

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Are you likely to add new elements to this enum in the foreseeable future? If no, use modelAsString=false instead.

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Are you likely to add new elements to this enum in the foreseeable future? If no, use modelAsString=false instead.

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Are any of these properties readonly?

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Are any of these properties readonly?

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: This seems like a good place for modelAsString: true since there are so many options you might add more soon. Others seem more like modelAsString: false is better

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: This should be less generic. Maybe `RecommendedActionState`

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: `RecommendedActionInitiatedBy` instead?

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: `RecommendedActionInitiatedBy` instead maybe? And modelAsString: false

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: It's not an encryption protector :)

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Is "create" correct? Isn't the location always equal to the server location?

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Is this required for PATCH?

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Required for patch?

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: encryption protector

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: arm-sql/2015-05-01-preview/examples/DatabaseRecommendedActionGet.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/examples/DatabaseRecommendedActionGet.json]: Recommended action names are always like that in production, so I would not simplify it

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: That's the only thing that can be updated

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: All of properties, except autoExecuteStatus, are read only

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Yep, that's only field used in patch

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: That's replicated from ArmSqlResource

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: This is generated by Arm.Tools

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Generated by Arm.Tools

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: it is just getting one advisor so this should be `Databases_GetAdvisor`

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: `Databases_UpdateAdvisor`

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: `Databases_GetRecommendedAction`

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: `Databases_UpdateRecommendedAction`

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: 👍

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: RecommendedActionInitiatedBy

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: RecommendedActionInitiatedBy

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Need x-ms-enum here, see https://travis-ci.org/Azure/azure-rest-api-specs/jobs/248011353

**Comment by sarangan12** [comment]
The CI is passing apart from few warnings. Please let me know if you plan to fix the names.

**Review by sarangan12 — APPROVED** [review]

**Comment by jaredmoo** [comment]
Thanks @sarangan12 , there are some comments related to required params that @tomapaunovic  is working on, please wait for my signoff before merging. Thanks!

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Done

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Done

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Done

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Done

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Done

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Done

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Done

File: arm-sql/2015-05-01-preview/swagger/advisors.json
Comment (tomapaunovic) [inline arm-sql/2015-05-01-preview/swagger/advisors.json]: Done

**Review by jaredmoo — APPROVED** [review]

**Comment by salameer** [comment]
@sarangan12, @jaredmoo seemed to have approved the changes. if we're all good here, lets pull the trigger on it

**Pull request merged**
